js/modules/k6/grpc/client.go

```diff
@@ -243,65 +295,18 @@ func (c *Client) Connect(ctxPtr *context.Context, addr string, params map[string
 		}
 	}
 
+	ctx, cancel := context.WithTimeout(c.vu.Context(), timeout)
+	defer cancel()
+
 	// (rogchap) Even with FailOnNonTempDialError, if there is a TLS error this will timeout
 	// rather than report the error, so we can't rely on WithBlock. By running in a goroutine
 	// we can then wait on the error channel instead, which could happen before the Dial
 	// returns. We only need to close the channel to un-block in a non-error scenario;
 	// otherwise it can be GCd without closing as we return on an error on the channel.
 	errc := make(chan error, 1)
-	go func() {
-		opts := []grpc.DialOption{
-			grpc.WithBlock(),
-			grpc.FailOnNonTempDialError(true),
-			grpc.WithStatsHandler(c),
-		}
-
-		if ua := state.Options.UserAgent; ua.Valid {
-			opts = append(opts, grpc.WithUserAgent(ua.ValueOrZero()))
-		}
-
-		if !isPlaintext {
-			tlsCfg := state.TLSConfig.Clone()
-			tlsCfg.NextProtos = []string{"h2"}
-
-			// TODO(rogchap): Would be good to add support for custom RootCAs (self signed)
-
-			// (rogchap) we create a wrapper for transport credentials so that we can report
-			// on any TLS errors.
-			creds := transportCreds{
-				credentials.NewTLS(tlsCfg),
-				errc,
-			}
-			opts = append(opts, grpc.WithTransportCredentials(creds))
-		}
+	defer close(errc)
```

Comment:
It seems to me like you can move the whole `errc` and just return an `err` from `c.dial` while wrapping all the goroutine and channel code inside of it.